applications/asset_tracker/src/cloud_codec/service_info.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <stdio.h>
+#include "service_info.h"
+
+#define SERVICE_INFO_JSON_NAME "serviceInfo"
+#define UI_JSON_NAME "ui"
+#define FOTAS_JSON_NAME "fota_v"
+#define FOTAS_JSON_NAME_SIZE (sizeof(FOTAS_JSON_NAME) + 5)
+
+static int add_array_obj(const char * const items[], const u32_t item_cnt,
+			 const char * const item_name, cJSON * const obj)
+{
+	cJSON *obj_to_add = NULL;
+
+	if ((obj == NULL) || (item_name == NULL)) {
+		return -EINVAL;
+	}
+
+	obj_to_add = cJSON_CreateArray();
+	if (obj_to_add == NULL) {
+		return -ENOMEM;
+	}
+
+	for (u32_t cnt = 0; cnt < item_cnt; ++cnt) {
+		if (items[cnt] != NULL) {
+			cJSON_AddItemToArray(obj_to_add,
+					     cJSON_CreateString(items[cnt]));
```

Comment:
`cJSON_CreateString` can fail here without being handled. We should probably do a `NULL` check, clean up `obj_to_add` and `return -ENOMEM`.